### PR TITLE
Add statistics route for execution environments

### LINKED ISSUE #24: Design and implement statistical route about pool size
As an administrator on CodeOcean  
I would like to get an overview of allocations/runner currently used or prewarmed
in order to get an impression of the overall system load.

<hr>

On CodeOcean, we have some statistics for each execution environment. We should design an API and implement it in Poseidon (and DCP afterward) to provide the information required. I assume the following information is useful per execution environment:

- Desired pre-warming pool size 
- Actual pre-warming pool size 
- Used runner pool size (as reserved for specific users and not available for new users)

This is how it the dashboard looks like right now (not all information included yet):

Bildschirmfoto 2021-11-01 um 14 07 43

### PR DESCRIPTION
**Pull request by mpass99**
Closes #24

Please push me if you see a use case for an additional test ;)

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -167,7 +212,7 @@ paths:
                       properties:
                         exposedPort:
                           description: The port inside the container.
-                          type: uint
+                          type: number
```
Comment (MrSerth) [inline api/swagger.yaml:215]: Is this an intended change here?

File: internal/api/api.go [hunk 2]
```diff
@@ -73,3 +79,16 @@ func Version(writer http.ResponseWriter, _ *http.Request) {
 		writer.WriteHeader(http.StatusNotFound)
 	}
 }
+
+// StatisticsExecutionEnvironments handles the route for statistics about execution environments.
+// It responds the prewarming pool size and the amount of idle runners and used runners.
```
Comment (MrSerth) [inline internal/api/api.go:84]: It's probably the "number" of runners as they are countable, isn't it?

**Review by MrSerth — COMMENTED** [review]
Looks good so far :) I'll try to implement support for the new route in CodeOcean soon and provide another update here ;)

**Review by MrSerth — CHANGES_REQUESTED** [review]
I implemented support for the new route in the following PR: https://github.com/openHPI/codeocean/pull/1105

While doing so, I noticed a small mistake. Otherwise, I like the new feature 🎉

File: api/swagger.yaml (hunk 1 above)
Comment (mpass99) [reply api/swagger.yaml:215]: Intended yes, but not as correct as it should be

File: api/swagger.yaml [hunk 3]
```diff
@@ -167,7 +212,8 @@ paths:
                       properties:
                         exposedPort:
                           description: The port inside the container.
-                          type: uint
+                          type: integer
+                          minimum: 0
```
Comment (MrSerth) [inline api/swagger.yaml:216]:
```suggestion
                          minimum: 0
                          maximum: 65535
```

We might also specify the maximum port number but don't need to.

**Review by MrSerth — APPROVED** [review]
Thanks for the (small) fix in the name of the JSON attributes. This will now work as expected for CodeOcean 🙂

File: api/swagger.yaml (hunk 1 above)
Comment (MrSerth) [reply api/swagger.yaml]: Ah, I see. Thanks for pointing that out!

**Pull request merged**